src/create-element.js

```diff
@@ -10,32 +9,38 @@ import { assign } from './util';
  * @returns {import('./internal').VNode}
  */
 export function createElement(type, props, children) {
-	props = assign({}, props);
+	let normalizedProps = {},
+		i;
+	for (i in props) {
+		if (i !== 'key' && i !== 'ref') normalizedProps[i] = props[i];
+	}
 
 	if (arguments.length > 3) {
 		children = [children];
 		// https://github.com/preactjs/preact/issues/1916
-		for (let i = 3; i < arguments.length; i++) {
+		for (i = 3; i < arguments.length; i++) {
 			children.push(arguments[i]);
 		}
 	}
 	if (children != null) {
-		props.children = children;
+		normalizedProps.children = children;
 	}
 
-	// "type" may be undefined during development. The check is needed so that
-	// we can display a nice error message with our debug helpers
-	if (type != null && type.defaultProps != null) {
-		for (let i in type.defaultProps) {
-			if (props[i] === undefined) props[i] = type.defaultProps[i];
+	// If a Component VNode, check for and apply defaultProps
+	// Note: type may be undefined in development, must never error here.
+	if (typeof type === 'function' && type.defaultProps != null) {
+		for (i in type.defaultProps) {
+			if (typeof normalizedProps[i] === 'undefined')
```

Comment:
can't we just do === undefined or is typeof cheaper?

Reply:
I believe in this case we can use === yes, I actually was testing this and forgot to remove it.